### PR TITLE
added textbox generator

### PR DESCRIPTION
**Pull request by supreethavadhani**
## Add a textbox generator

The commit consists of rendering all the textboxes in the form using a single line of html. See screenshots below.

### Ticket - [LAB-176](https://oregonstate-innovationlab.atlassian.net/browse/LAB-176)

| Before  | After |
| ----------- | ----------- |
| Previously a developer had to write 6 lines of html to render all the textboxes in the form below.      | After this PR, a developer can render all the textfields in a form with a single line of HTML code.     |
| Screen Shot 2022-06-27 at 8 49 53 PM   | Screen Shot 2022-06-27 at 10 46 12 PM     |

## What you need to check for

Screen Shot 2022-06-27 at 10 46 36 PM

1. The application runs on localhost 
2. The field validations work

### CODE REVIEW

File: src/app/framework-modules/elements/ct-field-generator/component.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { Component, Input, Output, EventEmitter} from '@angular/core';
+import {FormBuilder, FormGroup, Validators, FormControl} from '@angular/forms';
```
Comment (wbraynen) [inline src/app/framework-modules/elements/ct-field-generator/component.ts]: Formatting nitpick.  You have a space after `{` and a space before `}` on the neighboring lines, but not here.  Reformat/lint this line for consistency?

File: src/app/framework-modules/elements/ct-field-generator/component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+
+
+@Component({
+	selector: 'ct-field-generator',
+	templateUrl: './component.html',
+	styleUrls: ['./component.scss']
+})
+
+export class CTFieldGen {
```
Comment (wbraynen) [inline src/app/framework-modules/elements/ct-field-generator/component.ts]: Would it make sense to add a javadoc here to explain what this does?  Also, what does the "CT" prefix mean in "CTFieldGen"?  Finally, what is "Gen"?  Generator?  Maybe write it out in full, so it's more self documenting?

File: src/app/framework-modules/elements/elements.ts [hunk 3]
```diff
@@ -224,12 +226,14 @@ import { CtHorizontalBarChartComponent } from './ct-horizontal-barchart/componen
...
         /**
          * Miscellaneous Exports
          */
         ReactiveFormsModule,
         CommonModule,
         FormsModule,
+        
     ]
 })
   export class ElementsModule {
```
Comment (wbraynen) [inline src/app/framework-modules/elements/elements.ts]: I thought you had a linter installed?  If you do and your linter catches these things, but you forgot to run this new code through the linter, then I would highly recommend installing a linter as part of your PR checks.  Your linter could do one of the following:
Option 1. The linter lints and reformats the code up on the CI server and recommits it somehow. This is the more aggressive option.
Option 2. The linter lints (up on the CI server) and if the git repo is dirty after that (that is, the linter has made any changes), then your PR check fails. This is the more common conservative option.

**Review by wbraynen — CHANGES_REQUESTED** [review]
A few small nitpicky comments to address. Maybe time to create a PR check, at least for linting (if not to run unit tests)?  Also, would it make sense to have unit tests for this code?  Is it possible?

**Comment by wbraynen** [comment]
One other thing, though too late for this PR: as a work habit, you could prefix your branch name with the ticket number.  It then makes it easier to write the PR comments since to remember the ticket number all you have to do is look at your branch name. For example, `LAB-176-textbox-gen` instead of `textbox-gen`.  Also, FYI: snake case is easier to read than hyphens—studies show faster reading speed.

**Comment by osu-mp** [comment]
My feedback is not for this PR per se, but for the general project: what is meta-dev? I'm jumping in without the background that others may have, but it's not clear what the overall goal of the project is. The README has useful info on how to install/run the project, but a few sentences about what the project does/is for would be helpful. Alternatively, a link to a wiki with more info would work too.

**Comment by supreethavadhani** [comment]
@osu-mp created a new issue [Issue 2](https://github.com/supreethavadhani/metadev-client-example/issues/2)

File: src/app/framework-modules/elements/ct-field-generator/component.ts [hunk 4]
```diff
@@ -0,0 +1,31 @@
+import { Component, Input } from '@angular/core';
+import { FormControl} from '@angular/forms';
+import { Form } from '../../formdata/form/form';
+import { FormData } from '../../formdata/form/formData';
+
+
+
```
Comment (gangamamulya) [inline src/app/framework-modules/elements/ct-field-generator/component.ts]: There are many extra spaces, may be linting should be done to code.

**Review by gangamamulya — COMMENTED** [review]
Overall, the changes made seems to be very useful with the use of  ct-field-generator. And may be you can make it more self documenting and also add linting to format your code and remove extra spaces.

**Comment by wbraynen** [comment]
@supreethavadhani It looks like you're now using a mix of GitHub's issue track and Jira.  For example, this is not Jira: https://github.com/supreethavadhani/metadev-client-example/issues/2  Wouldn't using two different issue tracking platforms get confusing?

**Review by wbraynen — CHANGES_REQUESTED** [review]
What is a textbox generator?

The PR description is a little cryptic.  Maybe something like the following (see my screenshot below), plus an explanation of what the last screenshot illustrates?

Plus, either inline, some link to some context about what this is still?
______
image

**Comment by supreethavadhani** [comment]
> Overall, the changes made seems to be very useful with the use of ct-field-generator. And may be you can make it more self documenting and also add linting to format your code and remove extra spaces.

@gangamamulya I fixed the liniting issue. 
RE self documenting code - Im not able to figure out where I missed out on implementing self documenting code. 🤔  Am I missing something 👀  ! If you can help me find out where exactly I missed it I'll fix it and push a new commit 👍

**Comment by supreethavadhani** [comment]
> What is a textbox generator?
> 
> The PR description is a little cryptic. Maybe something like the following (see my screenshot below), plus an explanation of what the last screenshot illustrates?
> 
> Plus, either inline, some link to some context about what this is still?
> 
> image

Does this version look better?

**Comment by osu-mp** [comment]
Linter results look better. The other feedback I'd offer is seconding the suggestion for self-documenting code. Within the code, it's not obvious what field-generator does/why it reduces several lines into one.

**Review by wbraynen — CHANGES_REQUESTED** [review]
Please make more self-documenting.  For examples of what that means, see the comments I left on https://github.com/supreethavadhani/metadev-client-example/pull/3

**Comment by supreethavadhani** [comment]
The PR had a ton of file changes.
Reviewed code with @wbraynen in person.
Forced the merge because of the the 180,000 lines change.

**Pull request merged**